platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -241,6 +324,80 @@ def read_eeprom(self, offset, num_bytes):
         eeprom_raw = list(map(lambda h: int(h, base=16), eeprom_raw))
         return bytearray(eeprom_raw)
 
+    def read_eeprom(self, offset, num_bytes):
+        """
+        Read eeprom specfic bytes beginning from a random offset with size as num_bytes
+        Returns:
+            bytearray, if raw sequence of bytes are read correctly from the offset of size num_bytes
+            None, if the read_eeprom fails
+        Example:
+        mlxreg -d /dev/mst/mt52100_pciconf0 --reg_name MCIA --indexes slot_index=0,module=1,device_address=148,page_number=0,i2c_device_address=0x50,size=16,bank_number=0 -g
+        Sending access register...
+        Field Name            | Data
+        ===================================
+        status                | 0x00000000
+        slot_index            | 0x00000000
+        module                | 0x00000001
+        l                     | 0x00000000
+        device_address        | 0x00000094
+        page_number           | 0x00000000
+        i2c_device_address    | 0x00000050
+        size                  | 0x00000010
+        bank_number           | 0x00000000
+        dword[0]              | 0x43726564
+        dword[1]              | 0x6f202020
+        dword[2]              | 0x20202020
+        dword[3]              | 0x20202020
+        dword[4]              | 0x00000000
+        dword[5]              | 0x00000000
+        ....
+        16 bytes to read from dword -> 0x437265646f2020202020202020202020 -> Credo
+        """
+        # recalculate offset and page. Use 'ethtool' if there is no need to read vendor pages
+        if offset < SFP_VENDOR_PAGE_START:
+            return self._read_eeprom_specific_bytes(offset, num_bytes)
```

Comment:
What will be if offset + num_bytes > SFP_VENDOR_PAGE_START?

Reply:
**ethtool** will return bytes  SFP_VENDOR_PAGE_START - offset
For e.g  offset = 636 , num_bytes = 8, 
return 640 - 636 = 4 bytes
Suggest to handle it?
because mlxreg will return error